Formula/fish.rb

```diff
@@ -22,9 +22,16 @@ class Fish < Formula
 
   def install
     system "autoconf" if build.head? || build.devel?
+    args = %W[
+      --prefix=#{prefix}
+      --with-extra-functionsdir=#{HOMEBREW_PREFIX}/share/fish/vendor_functions.d
```

Comment:
`#{pkgshare}` will not work. The patch is about having version independent dirs outside of the Cellar so other Formulae can install completions and functions into that dir and fish can use those.
By default fish looking only into own Cellar share dirs for scripts. My PR is trying to make fish to look into #{HOMEBREW_PREFIX}/share/fish for completions and functions.

zsh has similar thing
`-enable-site-fndir=#{HOMEBREW_PREFIX}/share/zsh/site-functions`
and `zsh_completion.install` will install files into that dir.


I guess ZSH does not create site-functions dir during install and does not trigger the failure.